### ISSUE TITLE
Investigate State Management Tools for Next.js

### ISSUE THREAD

**Issue by qisforq** [AthletiFi/athletifi-website#212]
### Objective:
Research and evaluate various state management tools to determine the best fit for the Next.js app, considering their compatibility, performance, and ease of integration.

## Tools to Investigate:
1. Redux
2. Built-in React Features (Context API and Hooks)
3. MobX
4. Recoil

## Evaluation Criteria:
 - Assess how well each state management tool integrates with Next.js.
 - Evaluate the ease of learning and implementing each tool.
 - Assess the impact on rendering speed, memory usage, and overall application performance.
 - Evaluate ecosystem and community support:
 - Consider factors like available plugins, integrations, and active development.

## Deliverables:
- [ ] A lengthy, detailed, concise report summarizing the key findings and comparisons of the investigated state management tools.
- [ ] Recommendations on the most suitable state management tool based on the evaluation criteria and project requirements.
- [ ] Code examples or proof-of-concept implementations demonstrating the integration of the recommended tool with Next.js.

**Comment by HectorAgudelo** [comment]
Certainly! A simpler example using Jotai for state management in a Next.js project with TypeScript can be demonstrated through a basic counter application. This setup will involve creating a counter atom, and then using this atom in several components or pages to increment and display the counter value. Here’s how you can set it up:

### Step 1: Setup Your Next.js Project with TypeScript

If you haven't already, create a new Next.js project with TypeScript:

```bash
npx create-next-app my-next-app --typescript
cd my-next-app
```

### Step 2: Install Jotai

Install Jotai in your project:

```bash
npm install jotai
```

### Step 3: Define Jotai Atom

Create a Jotai atom to manage the counter state. You can put this in a `state` directory or directly in a file named `atoms.ts`.

```typescript
// state/atoms.ts
import { atom } from 'jotai';

export const counterAtom = atom(0);  // Initial value is 0
```

### Step 4: Create Components Using the Atom

Now, create components that will use this counter atom. For instance, let's make a component to display the counter and another to increment it.

#### Counter Display Component

```typescript
// components/CounterDisplay.tsx
import { useAtom } from 'jotai';
import { counterAtom } from '../state/atoms';

const CounterDisplay = () => {
  const [count] = useAtom(counterAtom);

  return <p>Counter: {count}</p>;
}

export default CounterDisplay;
```

#### Increment Button Component

```typescript
// components/IncrementButton.tsx
import { useAtom } from 'jotai';
import { counterAtom } from '../state/atoms';

const IncrementButton = () => {
  const [count, setCount] = useAtom(counterAtom);

  return <button onClick={() => setCount(count + 1)}>Increment</button>;
}

export default IncrementButton;
```

### Step 5: Use Components in Pages

Finally, integrate these components into one of your pages. You can use the default `index.tsx` page for this purpose.

```typescript
// pages/index.tsx
import CounterDisplay from '../components/CounterDisplay';
import IncrementButton from '../components/IncrementButton';

const HomePage = () => {
  return (
    <div>
      <h1>Welcome to the Counter App</h1>
      <CounterDisplay />
      <IncrementButton />
    </div>
  );
}

export default HomePage;
```

### Running Your Application

Run your Next.js application:

```bash
npm run dev
```

This setup will display a page with a counter and a button. Clicking the button will increment the counter, and the new value will be reflected in the display component. This demonstrates how Jotai can manage state across multiple components in a Next.js application using TypeScript.

**Comment by HectorAgudelo** [comment]
To distribute the counter state across multiple components in a Next.js application using Jotai, you can follow the same setup but extend the interaction between more components. Let's create an example where the counter state is shared and can be modified from three different components or pages.

### Step 1 to Step 4

These steps remain the same as in the previous example where you set up your Next.js project, install Jotai, define the `counterAtom`, and create basic components for displaying and modifying the counter.

### Step 5: Create More Components

To demonstrate the counter state being shared and modified across different components, let's add two more components that interact with the `counterAtom`.

#### Decrement Button Component

This component will decrease the counter value.

```typescript
// components/DecrementButton.tsx
import { useAtom } from 'jotai';
import { counterAtom } from '../state/atoms';

const DecrementButton = () => {
  const [count, setCount] = useAtom(counterAtom);

  return <button onClick={() => setCount(count - 1)}>Decrement</button>;
}

export default DecrementButton;
```

#### Counter Status Component

This component will display a message based on the counter value.

```typescript
// components/CounterStatus.tsx
import { useAtom } from 'jotai';
import { counterAtom } from '../state/atoms';

const CounterStatus = () => {
  const [count] = useAtom(counterAtom);
  const message = count > 0 ? "Positive" : count < 0 ? "Negative" : "Neutral";

  return <p>Counter Status: {message}</p>;
}

export default CounterStatus;
```

### Step 6: Use All Components in Pages

You can now incorporate all these components into your pages. For simplicity, let's use them in a single page, but they could be spread across different pages in a real application.

```typescript
// pages/index.tsx
import CounterDisplay from '../components/CounterDisplay';
import IncrementButton from '../components/IncrementButton';
import DecrementButton from '../components/DecrementButton';
import CounterStatus from '../components/CounterStatus';

const HomePage = () => {
  return (
    <div>
      <h1>Welcome to the Counter App</h1>
      <CounterDisplay />
      <IncrementButton />
      <DecrementButton />
      <CounterStatus />
    </div>
  );
}

export default HomePage;
```

### Summary

In this setup:

- The `CounterDisplay` component shows the current counter value.
- The `IncrementButton` and `DecrementButton` components allow the user to modify the counter value by incrementing and decrementing it, respectively.
- The `CounterStatus` component provides feedback based on the counter's current value.

This demonstrates how Jotai can be used to manage state effectively across multiple components, making it a good choice for state management in complex React and Next.js applications. This setup ensures that any changes to the counter state are immediately reflected across all components that use the `counterAtom`.
